PMMain/Midas/MidasGUI/RingData.cpp

```diff
@@ -0,0 +1,31 @@
+#include "RingData.h"
+
+
+ringData::ringData()
+{
+}
+
+
+ringData::~ringData()
+{
+}
+
+std::vector<ringData::keyboard_value> ringData::getRingInVector()
+{
+    return*ringInVector;
+}
```

Comment:
As in other comments, this should return the pointer, not the dereference object or else it can't be modified outside this class without a full "setRingInVector" callback (waste of time/space). The mutators don't really need to exist by this definition.